api/tests/Api/ResetPassword/ResetPasswordTest.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace App\Tests\Api\ResetPassword;
+
+use App\Entity\User;
+use App\Tests\Api\ECampApiTestCase;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+
+/**
+ * @internal
+ */
+class ResetPasswordTest extends ECampApiTestCase {
+    public function testPostResetPasswordResetsPasswordForAnonymousUser() {
+        /** @var User $user */
+        $user = static::$fixtures['user1manager'];
+
+        $this->createBasicClient()->request(
+            'POST',
+            '/auth/reset_password',
+            [
+                'json' => [
+                    'email' => $user->getEmail(),
+                ],
+            ]
+        );
+
+        $this->assertResponseStatusCodeSame(204);
+        self::assertEmailCount(1);
+        $mailerMessage = self::getMailerMessage();
+        self::assertEmailAddressContains($mailerMessage, 'To', $user->getEmail());
+        self::assertEmailHeaderSame($mailerMessage, 'subject', '[eCamp3] Password reset');
+
+        self::assertEmailHtmlBodyContains($mailerMessage, $user->getDisplayName());
+        self::assertEmailHtmlBodyContains($mailerMessage, 'http://localhost:3000/reset-password/');
+    }
+
+    /**
+     * @throws TransportExceptionInterface
+     * @throws ServerExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws ClientExceptionInterface
+     */
+    public function testPostResetPasswordResetsPasswordForUserLoggedInWithThisEmail() {
+        /** @var User $user */
+        $user = static::$fixtures['user1manager'];
+
+        $this->createClientWithCredentials(['email' => $user->getEmail()])
+            ->request(
+                'POST',
+                '/auth/reset_password',
+                [
+                    'json' => [
+                        'email' => $user->getEmail(),
+                    ],
+                ]
+            )
+        ;
+
+        $this->assertResponseStatusCodeSame(204);
+        self::assertEmailCount(1);
+    }
+
+    /**
+     * @throws TransportExceptionInterface
+     * @throws ServerExceptionInterface
+     * @throws RedirectionExceptionInterface
+     * @throws ClientExceptionInterface
+     */
+    public function testPostResetPasswordResetsPasswordForUserLoggedInWithAnotherEmail() {
```

Comment:
Does it make a difference whether the user is logged in? If so, does it make a difference if the user is logged in with a different email? And if so, do we need a test for that case as well?

Reply:
No sounds good. I was just wondering why you got the idea to add a separate test for logged in / not logged in when that makes no difference right now